render/egl.c

```diff
@@ -347,13 +369,113 @@ struct wlr_egl *wlr_egl_create(EGLenum platform, void *remote_display,
 		}
 	}
 
+	return true;
+}
+
+struct wlr_egl *wlr_egl_create(EGLenum platform, void *remote_display,
+		const EGLint *config_attribs) {
+	struct wlr_egl *egl = egl_create();
+	if (egl == NULL) {
+		wlr_log(WLR_ERROR, "Failed to create EGL context");
+		goto error;
+	}
+
+	if (!egl_init(egl, platform, remote_display, config_attribs)) {
+		wlr_log(WLR_ERROR, "Failed to initialize EGL context");
+		goto error;
+	}
+
 	return egl;
 
 error:
 	eglMakeCurrent(EGL_NO_DISPLAY, EGL_NO_SURFACE, EGL_NO_SURFACE, EGL_NO_CONTEXT);
 	if (egl->display) {
 		eglTerminate(egl->display);
 	}
+	if (egl) {
+		free(egl);
```

Comment:
Since `free(NULL)` is a no-op, we don't need the `if` here.

Reply:
It seems I was calling `goto error` if `egl_create` fails, which was a bit dumb. I changed the error handling logic in `wlr_egl_create`.